### PR TITLE
SB-25552 - FusionAuth Deployment

### PR DESCRIPTION
**Pull request by shiva-rakshith**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-25552" title="SB-25552" target="_blank">StorySB-25552</a>  UCI - Deployment
  </td></table>
  <br />
 

fusionauth deploy configuration

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shiva-rakshith** [comment]
@keshavprasadms Please review the PR.

File: ansible/roles/stack-sunbird/defaults/main.yml [hunk 1]
```diff
@@ -278,6 +278,7 @@ service_env:
     - ../../../../ansible/roles/stack-sunbird/templates/transformer_industry_feedback.xml
   orchestrator:
     - ../../../../ansible/roles/stack-sunbird/templates/orchestrator.env
+  fusionauth: ../../../../ansible/roles/stack-sunbird/templates/fusionauth.env
```
Comment (keshavprasadms) [inline ansible/roles/stack-sunbird/defaults/main.yml:281]: Why do we need an empty env file?

File: kubernetes/helm_charts/core/fusionauth/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "fusionauth.fullname" . -}}
+{{- $ingressPaths := .Values.ingress.paths -}}
+{{- $extraPaths := .Values.ingress.extraPaths -}}
+apiVersion: {{ include "fusionauth.ingressApiVersion" . }}
+kind: Ingress
```
Comment (keshavprasadms) [inline kubernetes/helm_charts/core/fusionauth/templates/ingress.yaml:6]: We dont have Ingress in our Kube cluster

File: kubernetes/pipelines/deploy_uci/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+               sh'git clean -fxd'
+            }  
+            checkout scm
+        }
+
+        stage('deploy') {
+            envDir = sh(returnStdout: true, script: "echo $JOB_NAME").split('/')[-3].trim()
+	        module = sh(returnStdout: true, script: "echo $JOB_NAME").split('/')[-2].trim()
+	        jobName = sh(returnStdout: true, script: "echo $JOB_NAME").split('/')[-1].trim()
+		    values = [:]
```
Comment (keshavprasadms) [inline kubernetes/pipelines/deploy_uci/Jenkinsfile:24]: Please correct spacing of the lines

File: ansible/roles/stack-sunbird/defaults/main.yml (hunk 1 above)
Comment (shiva-rakshith) [reply ansible/roles/stack-sunbird/defaults/main.yml:281]: We are using 'sunbird-deploy' job, it is throwing an error, if we don't have env file.

File: kubernetes/pipelines/deploy_uci/Jenkinsfile (hunk 3 above)
Comment (shiva-rakshith) [reply kubernetes/pipelines/deploy_uci/Jenkinsfile]: Corrected the spacing.

File: kubernetes/helm_charts/core/fusionauth/templates/ingress.yaml (hunk 2 above)
Comment (shiva-rakshith) [reply kubernetes/helm_charts/core/fusionauth/templates/ingress.yaml]: Removed ingress.yaml and related references.

File: ansible/roles/stack-sunbird/defaults/main.yml (hunk 1 above)
Comment (keshavprasadms) [reply ansible/roles/stack-sunbird/defaults/main.yml:281]: We have role named `helm-deploy` which is meant for such charts where we dont need env. Did you try with that role?

Comment (shiva-rakshith) [reply ansible/roles/stack-sunbird/defaults/main.yml:281]: No, I will check.

Comment (shiva-rakshith) [reply ansible/roles/stack-sunbird/defaults/main.yml]: Tested with helm-deploy, deployment is successfully done. Removed env file and related changes.

**Review by keshavprasadms — APPROVED** [review]

**Pull request merged**